lib/wallet/src/Cardano/Wallet/DB/Store/Wallets/Model.hs

```diff
@@ -93,6 +98,24 @@ garbageCollectTxWalletsHistory (TxSet txh, mtxmh) = (TxSet (gc txh), mtxmh)
     gc :: Map TxId x -> Map TxId x
     gc x = Map.restrictKeys x $ walletsLinkedTransactions mtxmh
 
+-- | List of transactions that are to be deleted from the 'TxSet'
+-- when rolling back the collection of wallets.
+--
+-- These are precisely those transactions that have been rolled
```

Comment:
Isn't this a path to a live lock ? If 2 wallets share a tx they will not delete it because the other wallet needs it . We need to make sure they do not end up in this code at the same time.